src/features/checkout/screen/index.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useState } from 'react';
+import { ScrollView, Switch, Text, TextInput, TouchableOpacity, View } from 'react-native';
+
+import { Picker } from '@react-native-picker/picker';
+import { NavigationProp } from '@react-navigation/native';
+
+import { translate } from 'localization/hooks';
+
+import { useAddOrder } from 'network/queries/order-queries';
+
+import styles from './styles';
+
+type CheckoutScreenProps = {
+  navigation: NavigationProp<any>;
+};
+
+const CheckoutScreen: React.FC<CheckoutScreenProps> = props => {
+  const [error, setError] = useState('');
+  const [city, setCity] = useState('');
+  const [country, setCountry] = useState('');
+  const [addressLine1, setAddressLine1] = useState('');
+  const [addressLine2, setAddressLine2] = useState('');
+  const [postalCode, setPostalCode] = useState('');
+  const [state, setState] = useState('');
+  const [cardNumber, setCardNumber] = useState('');
+  const [selectedMonth, setSelectedMonth] = useState(1);
+  const [selectedYear, setSelectedYear] = useState(2024);
+  const [cvcCode, setCvcCode] = useState('');
+  const [sameBillingAsShipping, setSameBillingAsShipping] = useState(true);
+
+  const [billingCity, setBillingCity] = useState('');
+  const [billingCountry, setBillingCountry] = useState('');
+  const [billingAddressLine2, setBillingAddressLine2] = useState('');
+  const [billingAddressLine1, setBillingAddressLine1] = useState('');
+  const [billingPostalCode, setBillingPostalCode] = useState('');
+
+  const months = Array.from({ length: 12 }, (_, i) => ({ label: `${i + 1}`, value: i + 1 }));
+  const currentYear = new Date().getFullYear();
+  const years = Array.from({ length: 20 }, (_, i) => ({
+    label: `${currentYear + i}`,
+    value: currentYear + i,
+  }));
```

Comment:
All these + the validation lines in the `handleSubmit` method could be saved by using `react-hook-form` https://react-hook-form.com/

Reply:
nice! I'll take a look at it